data/trie/trieStorageManager.go

```diff
@@ -376,7 +376,8 @@ func (tsm *trieStorageManager) takeSnapshot(snapshot *snapshotsQueueEntry, msh m
 		return
 	}
 
-	err = newRoot.commit(true, 0, tsm.db, db)
+	maxTrieLevelInMemory := uint(5)
```

Comment:
here you need the config values as well.

Reply:
Not really, because after the snapshot is done, the pointer to the newRoot will not be used again, so it will not influence the trie in any way.